File: src/modules/authentication/services/auth_refresh_service.py
import os
from src.modules.authentication.domain.dto.auth_refresh_dto import (
    AuthRefreshDTO,
)
from src.modules.user.models import User
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.services.token_verify_service import TokenVerifyService
from src.shared.validators.status_validator import status_validator
from src.shared.services.token_sign_service import TokenSignService
from src.shared.domain.enums.period_expiration_enum import PeriodExpiration


class AuthRefreshService:
    def execute(dto: AuthRefreshDTO):
        payload = TokenVerifyService.execute(
            token=dto.refresh_token, secret=os.getenv("JWT_REFRESH_SECRET", "")
        )

        user = User.objects.filter(email=payload["sub"], deleted_at=None).first()

        if user is None:
            raise NotFoundException("Usuário não encontrado!")

        status_validator(user.status)

        return {
            "user": user,
            "token": TokenSignService.execute(user=user, time=10),
            "refresh_token": TokenSignService.execute(
                user=user,
                time=10,
                period=PeriodExpiration.DAY,
                secret=os.getenv("JWT_REFRESH_SECRET", ""),
            ),
        }

File: src/main/java/br/com/guizzo/projectbasic/modules/auth/services/AuthRefreshService.java
package br.com.guizzo.projectbasic.modules.auth.services;

import br.com.guizzo.projectbasic.modules.auth.domain.dto.AuthRefreshDTO;
import br.com.guizzo.projectbasic.modules.auth.domain.dto.AuthRefreshResponseDTO;
import br.com.guizzo.projectbasic.modules.user.domain.entities.User;
import br.com.guizzo.projectbasic.modules.user.repositories.UserRepository;
import br.com.guizzo.projectbasic.shared.domain.enums.PeriodExpiration;
import br.com.guizzo.projectbasic.shared.exceptions.NotFoundException;
import br.com.guizzo.projectbasic.shared.services.TokenSignService;
import br.com.guizzo.projectbasic.shared.services.TokenVerifyService;
import br.com.guizzo.projectbasic.shared.validators.StatusValidator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class AuthRefreshService {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private TokenSignService tokenSignService;

    @Autowired
    private TokenVerifyService tokenVerifyService;

    @Value("${JWT_REFRESH_SECRET}")
    private String refreshSecret;

    public AuthRefreshResponseDTO execute(AuthRefreshDTO dto){

        String payload = tokenVerifyService.execute(dto.getRefreshToken(), refreshSecret);

        User user = userRepository.findFirstByEmailAndDeletedAtIsNull(payload);
        if(user == null) throw new NotFoundException("Usuário não encontrado!");

        StatusValidator.validate(user.getStatus());

        String token = tokenSignService.execute(user);
        String refreshToken = tokenSignService.execute(user, 10, PeriodExpiration.DAY, refreshSecret);

        return new AuthRefreshResponseDTO(
                user,
                token,
                refreshToken
        );

    }
}
